lib/board.rb

```diff
@@ -65,4 +65,40 @@ def render(show_ships_boolean = false)
     "C #{["C1", "C2", "C3", "C4"].map { |coord| @cells[coord].render(show_ships_boolean)}.join(" ")} \n" +
     "D #{["D1", "D2", "D3", "D4"].map { |coord| @cells[coord].render(show_ships_boolean)}.join(" ")} \n"
   end
+
+  def random_coord
+    @cells.keys.sample
+  end
+
+  def random_cell
+    @cells[random_coord]
+  end
+
+  def place_ship_randomly(ship_object)
+    until ship_object.length == @cells.values.count{|cell| cell.ship == ship_object}
+
+    # get a random coordinate
+      initial_coord = random_coord
+      coord_letter = initial_coord.chars[0]
+      coord_number = initial_coord.chars[1]
```

Comment:
Since we use this in multiple places, maybe this is a good candidate for breaking out into a helper method